pkg/controller/etcdproxy/etcdproxy_controller.go

```diff
@@ -259,81 +301,71 @@ func (c *Controller) syncHandler(key string) error {
-	//if etcdstorage.Spec.Replicas != nil && *etcdstorage.Spec.Replicas != *deployment.Spec.Replicas {
-	//	glog.V(4).Infof("EtcdStorage %s replicas: %d, deployment replicas: %d", name, *etcdstorage.Spec.Replicas, *deployment.Spec.Replicas)
-	//	deployment, err = c.kubeclientset.AppsV1().Deployments(etcdstorage.Namespace).Update(newDeployment(etcdstorage))
-	//}
-	//
-	//// If an error occurs during Update, we'll requeue the item so we can
-	//// attempt processing again later. THis could have been caused by a
-	//// temporary network failure, or any other transient reason.
-	//if err != nil {
-	//	return err
-	//}
-
-	//// Finally, we update the status block of the EtcdStorage resource to reflect the
-	//// current state of the world
-	//err = c.updateEtcdStorageStatus(etcdstorage, deployment)
-	//if err != nil {
-	//	return err
-	//}
+	replicaset, err := c.replicasetsLister.ReplicaSets(c.controllerNamespace).Get(getReplicaSetName(etcdstorage))
+	if errors.IsNotFound(err) {
+		replicaset, err = c.kubeclientset.AppsV1().ReplicaSets(c.controllerNamespace).Create(c.newReplicaSet(etcdstorage))
+	}
+
+	// If an error occurs during Get/Create, we'll requeue the item so we can
+	// attempt processing again later. This could have been caused by a
+	// temporary network failure, or any other transient reason.
+	if err != nil {
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+
+	// If the ReplicaSet is not controlled by this EtcdStorage resource, we should log
+	// a warning to the event recorder and ret
+	if !metav1.IsControlledBy(replicaset, etcdstorage) {
+		msg := fmt.Sprintf(MessageResourceExists, replicaset.Name)
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrResourceExists, msg)
+		return fmt.Errorf(msg)
+	}
+
+	// Create Service to expose the etcdproxy pod.
+	serviceName := fmt.Sprintf("etcd-%s", etcdstorage.ObjectMeta.Name)
+	service, err := c.servicesLister.Services(c.controllerNamespace).Get(serviceName)
+	if errors.IsNotFound(err) {
+		service, err = c.kubeclientset.CoreV1().Services(c.controllerNamespace).Create(c.newService(etcdstorage))
+	}
+
+	// If an error occurs during Get/Create, we'll requeue the item so we can
+	// attempt processing again later. This could have been caused by a
+	// temporary network failure, or any other transient reason.
+	if err != nil {
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrUnknown, err.Error())
+		return err
+	}
+
+	// If the Service is not controlled by this EtcdStorage resource, we should log
+	// a warning to the event recorder and ret
+	if !metav1.IsControlledBy(service, etcdstorage) {
+		msg := fmt.Sprintf(MessageResourceExists, service.Name)
+		c.recorder.Event(etcdstorage, corev1.EventTypeWarning, ErrResourceExists, msg)
+		return fmt.Errorf(msg)
+	}
+
+	// Finally, we update the status block of the EtcdStorage resource to reflect the
+	// current state of the world
+	err = c.updateEtcdStorageStatus(etcdstorage)
```

Comment:
still my question: does this cause a status update request even though nothing changed?

Reply:
If nothing changed, this should be a noop.

Reply:
As we're not changing the status of the EtcdStorage for now (status and conditions are to be implemented in #14), we'll remove this line.

Reply:
👍 , just put a TODO instead of it.